nifi-nar-bundles/nifi-smb-bundle/nifi-smb-client-provider-api/src/main/java/org/apache/nifi/services/smb/NiFiSmbClient.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.nifi.services.smb;
+
+import java.io.OutputStream;
+import java.util.stream.Stream;
+
+public interface NiFiSmbClient {
+
+    Stream<SmbListableEntity> listRemoteFiles(String path);
+
+    void createDirectory(String path);
+
+    OutputStream getOutputStreamForFile(String pathAndFileName);
```

Comment:
Recommend adjusting this method name and parameter:
```suggestion
    OutputStream getFileOutputStream(String path);
```